src/txn/txn_recover.c

```diff
@@ -695,8 +706,12 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
      * 2. A valid recovery timestamp. The recovery timestamp is the stable timestamp retrieved
      *    from the metadata checkpoint information to indicate the stable timestamp when the
      *    checkpoint happened. Anything updates newer than this timestamp must rollback.
+     * 3. The history store file was found in the metadata.
      */
-    if (!F_ISSET(conn, WT_CONN_READONLY) && conn->txn_global.recovery_timestamp != WT_TS_NONE) {
+    if (hs_does_not_exist)
```

Comment:
Why does not finding a history store file mean we skip the checkpoint after log recovery? i.e: shouldn't this just be folded into the first if, rather than pulled into a different clause? Regardless this pattern of having an empty if condition isn't usual in our code - it's better to include it in the condition checks that lead to code being executed.

Reply:
My mistake. We should checkpoint regardless.